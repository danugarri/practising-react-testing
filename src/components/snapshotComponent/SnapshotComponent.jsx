import React from 'react'
import { NavLink } from 'react-router-dom'

export const SnapshotComponent = () => {

  return (
    <React.Fragment>
        <h1>This is a component we do not want to be modified</h1>
        <span>
            According to the previus statement  we will create a SNAPSHOT TEST to avoid changes
        </span>
        <p>Please look at the code to find it</p><hr />
        <NavLink to = '/'>Home</NavLink>
    </React.Fragment>
  )
}
