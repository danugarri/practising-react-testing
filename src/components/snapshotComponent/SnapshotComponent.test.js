import renderer from 'react-test-renderer';
import { SnapShotComponent } from './SnapshotComponent';

describe('The SnapShotComponent', () => {
    
    it('renders correctly', () => {
        const tree = renderer
            .create(<SnapShotComponent />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

