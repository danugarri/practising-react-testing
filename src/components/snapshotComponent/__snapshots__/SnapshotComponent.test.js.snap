// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SnapShotComponent renders correctly 1`] = `
Array [
  <h1>
    This is a component we do not want to be modified
  </h1>,
  <span>
    According to the previus statement  we will create a SNAPSHOT TEST to avoid changes
  </span>,
  <p>
    Please look at the code to find it
  </p>,
  <hr />,
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>,
]
`;
